diff --git a/interface-definitions/system_console.xml.in b/interface-definitions/system_console.xml.in
index 5acd3e90b..6e0c4d064 100644
--- a/interface-definitions/system_console.xml.in
+++ b/interface-definitions/system_console.xml.in
@@ -12,13 +12,17 @@
             <properties>
               <help>Serial console device name</help>
               <completionHelp>
-                <script>ls -1 /dev | grep -e ttyS -e hvc</script>
+                <script>ls -1 /dev | grep -e ttyS -e hvc -e ttyAM</script>
                 <script>if [ -d /dev/serial/by-bus ]; then ls -1 /dev/serial/by-bus; fi</script>
               </completionHelp>
               <valueHelp>
                 <format>ttySN</format>
                 <description>TTY device name, regular serial port</description>
               </valueHelp>
+              <valueHelp>
+                <format>ttyAN</format>
+                <description>TTY device name, abstract serial port</description>
+              </valueHelp>
               <valueHelp>
                 <format>usbNbXpY</format>
                 <description>TTY device name, USB based</description>
@@ -28,7 +32,7 @@
                 <description>Xen console</description>
               </valueHelp>
               <constraint>
-                  <regex>(ttyS[0-9]+|hvc[0-9]+|usb[0-9]+b.*)</regex>
+                  <regex>(ttyS[0-9]+|ttyAMA?[0-9]+|hvc[0-9]+|usb[0-9]+b.*)</regex>
               </constraint>
             </properties>
             <children>
