diff --git a/src/conf_mode/container.py b/src/conf_mode/container.py
index 3636b0871..9d0982994 100755
--- a/src/conf_mode/container.py
+++ b/src/conf_mode/container.py
@@ -396,7 +396,7 @@ def generate_run_arguments(name, container_config):
             name_server += f'--dns {ns}'
 
     container_base_cmd = f'--detach --interactive --tty --replace {capabilities} --cpus {cpu_quota} {sysctl_opt} ' \
-                         f'--memory {memory}m --shm-size {shared_memory}m --memory-swap 0 --restart {restart} ' \
+                         f'--memory {memory}m --shm-size {shared_memory}m --restart {restart} ' \
                          f'--name {name} {hostname} {device} {port} {name_server} {volume} {tmpfs} {env_opt} {label} {uid} {host_pid}'
 
     entrypoint = ''
